include/deal.II/base/standard_tensors.h

```diff
@@ -0,0 +1,513 @@
+const SymmetricTensor<2, dim>
+StandardTensors::Elasticity<dim>::I = unit_symmetric_tensor<dim>();
+
+
+
+template <int dim>
+template <typename Number>
+inline
+Tensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::F (const Tensor<2, dim, Number> &Grad_u)
+{
+ // See Wriggers (2008) p 23 equ. 3.14
+ return I + Grad_u;
+}
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::C (const Tensor<2, dim, Number> &F)
+{
+ // See Wriggers (2008) p 23 equ. 3.15 (text below)
+ return symmetrize(transpose(F)*F);
+}
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::b (const Tensor<2, dim, Number> &F)
+{
+ // See Wriggers (2008) p 23 equ. 3.25 (text below)
+ return symmetrize(F*transpose(F));
+}
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::E (const Tensor<2, dim, Number> &F)
+{
+ // See Wriggers (2008) p 23 equ. 3.15
+ return Number(0.5)*(C(F) - unit_symmetric_tensor<dim,Number>());
+}
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::e (const Tensor<2, dim, Number> &Grad_u)
+{
+ // See Wriggers (2008) p 24 equ. 3.17
+ // This is the equivalent to 0.5*symmetrize(Grad_u + transpose(Grad_u));
+ return symmetrize(Grad_u);
+}
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+StandardTensors::Elasticity<dim>::a (const Tensor<2, dim, Number> &F)
+{
+ const Tensor<2, dim, Number> F_inv = invert(F);
+ // See Wriggers (2008) p 30 equ. 3.35
+ return Number(0.5)*symmetrize(unit_symmetric_tensor<dim,Number>() - transpose(F_inv)*F_inv);
+}
+
+
+
+template <int dim>
+const SymmetricTensor<4, dim>
+StandardTensors::Elasticity<dim>::II = identity_tensor<dim>();
+
+
+
+template <int dim>
+const SymmetricTensor<4, dim>
+StandardTensors::Elasticity<dim>::IxI = outer_product(I, I);
+
+
+
+template <int dim>
+const SymmetricTensor<4, dim>
+StandardTensors::Elasticity<dim>::dev_P = deviator_tensor<dim>();
+
+
+
+template <int dim>
+template <typename Number>
+inline
+SymmetricTensor<4, dim, Number>
+StandardTensors::Elasticity<dim>::Dev_P (const Tensor<2, dim, Number> &F)
+{
+  const Number det_F = determinant(F);
+  Assert(det_F > Number(0.0),
+         ExcMessage("Deformation gradient has a negative determinant."));
+  const Tensor<2,dim,Number> C_ns = transpose(F)*F;
+  const SymmetricTensor<2,dim,Number> C = symmetrize(C_ns);
+  const SymmetricTensor<2,dim,Number> C_inv = symmetrize(invert(C_ns));
+
+  // See Wriggers p46 equ 3.125
+  SymmetricTensor<4,dim,Number> Dev_P = outer_product(C_inv,C);  // Dev_P = C_inv_x_C
+  Dev_P /= -dim;                                                 // Dev_P = -[1/dim]C_inv_x_C
+  Dev_P += SymmetricTensor<4,dim,Number>(II);                    // Dev_P = II - [1/dim]C_inv_x_C
```

Comment:
Same here -- as long as `Number() + double()` makes sense, it should make sense to add `SymmetricTensor<4,dim,double>` to `Dev_P`.